src/codec/crypto/nonce.rs

```diff
@@ -18,7 +18,7 @@ impl Nonce {
     }
 
     pub fn from_bytes(data: &[u8; NONCE_LENGTH]) -> Self {
-        Self(data.clone())
+        Self(*data)
```

Comment:
Wow, super cool! I had no idea[ `Copy` is implemented for `[T;N]`](https://doc.rust-lang.org/std/marker/trait.Copy.html#impl-Copy-for-%5BT;+N%5D)